subprojects/core/src/main/java/org/gradle/internal/buildtree/BuildTreeScopeServices.java

```diff
@@ -69,4 +78,35 @@ protected ExceptionAnalyser createExceptionAnalyser(ListenerManager listenerMana
     public DefaultProjectStateRegistry createProjectPathRegistry(WorkerLeaseService workerLeaseService) {
         return new DefaultProjectStateRegistry(workerLeaseService);
     }
+
+    ValidateStep.ValidationWarningReporter createValidationWarningReporter() {
+        return new WorkValidationWarningReporterImpl();
+    }
+
+    private static class WorkValidationWarningReporterImpl implements ValidateStep.ValidationWarningReporter, Closeable {
+        private static final Logger LOGGER = LoggerFactory.getLogger(WorkValidationWarningReporterImpl.class);
+
+        private final AtomicInteger workWithFailuresCount = new AtomicInteger();
+
+        @Override
+        public void reportValidationWarnings(UnitOfWork work, Collection<String> warnings) {
+            workWithFailuresCount.incrementAndGet();
+            LOGGER.warn("Validation failed for {}, disabling optimizations:{}",
+                work.getDisplayName(),
+                warnings.stream().map(warning -> "\n  - " + warning).collect(Collectors.joining()));
+            warnings.forEach(warning -> DeprecationLogger.deprecateBehaviour(warning)
+                .withContext("Due to the failed validation execution optimizations are disabled.")
```

Comment:
```suggestion
                .withContext("Execution optimizations are disabled due to the failed validation.")
```

It took me three times to understand the message, probably because of the three `ions` together in the middle.

Reply:
I don't feel strongly, changed.